mixins/localize/demo/localize-mixin-sandbox.js

```diff
@@ -0,0 +1,348 @@
+		};
+	}
+
+	get message() {
+		return this._selectedTemplate.key === 'custom' ? this.shadowRoot.querySelector('#message').value : langResources[this._selectedTemplate.key];
+	}
+
+	connectedCallback() {
+		super.connectedCallback();
+		this._selectTemplate(this.selectedTemplate);
+	}
+
+	render() {
+
+		setTimeout(() => this.shadowRoot.querySelector('#code')?.forceUpdate(), 0);
+
+		const tags = this.variables.find(v => v.type === 'tag') && Object.entries(this.tags).reduce((acc, [k, v]) => (acc[k] = eval(v)) && acc, {});
+		const localizeMethod = tags ? 'localizeHTML' : 'localize';
+		const renderedArgs = Object.entries({ ...this.arguments, ...this.tags }).filter(([k, val]) => val && this.variables.find(v => v.name === k));
+
+		return html`		
+		<div id="actions">
+			<d2l-dropdown-button text="Load Template" primary>
+				<d2l-dropdown-menu>
+			    <d2l-menu label="Templates">${this.constructor.templates.map(({ name, key }) => html`
+			      <d2l-menu-item text="${name}" @click="${this.handleTemplateClick}" data-template="${key}"></d2l-menu-item>`)}
+					</d2l-menu>
+				</d2l-dropdown-menu>
+			</d2l-dropdown-button>
+		</div>
+		
+		<h2 class="d2l-heading-4">Message</h2>
+		<d2l-input-textarea id="message" label="Message" label-hidden @input="${this._setCustomTemplate}" value="${this.message}" max-rows="40" rows="3">
+		</d2l-input-textarea>
+
+		<div ?hidden="${!this.variables.length}">
+			<h2 class="d2l-heading-4">Arguments</h2>
+			<div>${this.variables.map(({ name, type, defaultTag }) => (type === 'tag' ? html`
+				<d2l-input-text name="${name}" label="&lt;${name}&gt;${ defaultTag ? ' (default tag)' : ''}" value="${this.arguments[name]}" @input="${this._setArgument}" type="${type}">
+					<d2l-button-icon slot="right" text="Generate a link" icon="tier1:link" @click="${this.handleGenerateLinkClick}"></d2l-button-icon>
+					<d2l-button-icon slot="right" text="Generate a tooltip" icon="tier1:messages" @click="${this.handleGenerateTooltipClick}"></d2l-button-icon>
+					<d2l-button-icon slot="right" text="Build a custom generator" icon="html-editor:source-editor" @click="${this.handleBuildGeneratorClick}"></d2l-button-icon>
+				</d2l-input-text>` : html`
+				<d2l-input-text type="${type}" name="${name}" label="{${name}}" value="${this.arguments[name]}" @input="${this._setArgument}"></d2l-input-text>`))}
+			</div>
+		</div>
+
+		<div ?hidden="${this._error}">
+			<h2 class="d2l-heading-4">Code</h2>
+			<d2l-code-view id="code" language="javascript">
+				this.${localizeMethod}('${this._selectedTemplate.key}'${renderedArgs.length ? `, {${renderedArgs.map(([k, val]) => `
+					${k}: ${val.constructor === String && !this.tags[k] ? `'${val.toString()}'` : val.toString()}`).join(',')}
+				}` : ''});
+			</d2l-code-view>
+		</div>
+
+		<h2 id="result" class="d2l-heading-4">
+			<span>Result</span>
+			<div>
+				<d2l-dropdown open-on-hover>
+					<d2l-icon id="results-help" icon="tier1:help" class="d2l-dropdown-opener"></d2l-icon>
```

Comment:
Not a huge deal since this isn't product code, but this likely isn't super accessible since the icon isn't really meant to be an opener/target for things like dropdowns as it's unlabelled. Probably a button-icon would be more appropriate.

Reply:
Ah, right, originally I had a tooltip on it, but you can't get to the link in a tooltip so I switched it out but hadn't thought of this aspect. It opens on hover, it that weird for a button?

Reply:
It's a little bit strange, but at least a button would have a `tabindex` so it would also open when it received focus.